### PR TITLE
ttl: add metrics for task manager

### LINKED ISSUE #40810: Add metrics about TTL tasks

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?

Issue Number: close #40810

Problem Summary:

Add a metric to track the owned task with different task status.

image

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### CODE REVIEW

File: ttl/ttlworker/BUILD.bazel [hunk 1]
```diff
@@ -55,6 +55,7 @@ go_test(
     ],
     embed = [":ttlworker"],
     flaky = True,
+    race = "on",
```
Comment (YangKeao) [inline ttl/ttlworker/BUILD.bazel:58]: @hawkingrei PTAL :handshake:

**Review by hawkingrei — APPROVED** [review]

File: ttl/ttlworker/task_manager.go [hunk 2]
```diff
@@ -474,6 +498,20 @@ func (m *taskManager) checkInvalidTask(se session.Session) {
...
+	for _, task := range m.runningTasks {
+		switch task.Status {
+		case cache.TaskStatusRunning:
+			runningTasks++
+		case cache.TaskStatusFinished:
+			finishedTask++
+		}
+	}
+	metrics.RunningTaskCnt.Set(runningTasks)
+	metrics.FinishedTaskCnt.Set(finishedTask)
```
Comment (lcwangchao) [inline ttl/ttlworker/task_manager.go:512]: Is it true that the finished task will be always true because it will be removed immediately once finished

**Review by lcwangchao — COMMENTED** [review]
I think we should add a new chart to granfana. Will it be added in another PR?

File: ttl/ttlworker/job_manager.go [hunk 3]
```diff
@@ -45,7 +45,7 @@ const setTableStatusOwnerTemplate = `UPDATE mysql.tidb_ttl_table_status
 	SET current_job_id = %?,
 		current_job_owner_id = %?,
 		current_job_start_time = %?,
-		current_job_status = 'waiting',
+		current_job_status = 'running',
```
[the remark this answers is not in the document]
Comment (YangKeao) [reply ttl/ttlworker/job_manager.go:48]: Yes. Without this modification, the job status will be always 'waiting'.

File: ttl/ttlworker/task_manager.go (hunk 2 above)
Comment (YangKeao) [reply ttl/ttlworker/task_manager.go]: Yes. I rewrote this function to set the count with `len(m.runningTasks)` directly.

**Comment by YangKeao** [comment]
> I think we should add a new chart to granfana. Will it be added in another PR?

Added now @lcwangchao

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**